src/smart-components/group/groups.js

```diff
@@ -1,49 +1,65 @@
 import React, { Fragment, useState, useEffect } from 'react';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import PropTypes from 'prop-types';
-import { Link, Route, Switch } from 'react-router-dom';
+import { Link, Route, Switch, useHistory } from 'react-router-dom';
 import { expandable } from '@patternfly/react-table';
 import { Button, Stack, StackItem } from '@patternfly/react-core';
 import AddGroupWizard from './add-group/add-group-wizard';
 import EditGroup from './edit-group-modal';
 import RemoveGroup from './remove-group-modal';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
 import { TableToolbarView } from '../../presentational-components/shared/table-toolbar-view';
 import { createRows } from './group-table-helpers';
 import { fetchGroups } from '../../redux/actions/group-actions';
 import Group from './group';
 import { TopToolbar, TopToolbarTitle } from '../../presentational-components/shared/top-toolbar';
-import { defaultSettings } from '../../helpers/shared/pagination';
 import { Section } from '@redhat-cloud-services/frontend-components';
 import Role from '../role/role';
 
 const columns = [{ title: 'Name', cellFormatters: [ expandable ]}, 'Description', 'Members', 'Last modified' ];
 
-const Groups = ({ fetchGroups, isLoading, pagination, history: { push }, groups, userIdentity }) => {
+const Groups = () => {
   const [ filterValue, setFilterValue ] = useState('');
   const [ selectedRows, setSelectedRows ] = useState([]);
   const [ removeGroupsList, setRemoveGroupsList ] = useState([]);
 
+  const dispatch = useDispatch();
+  const { groups, pagination, userIdentity, isLoading } = useSelector(({ groupReducer: { groups, isLoading }}) => ({
```

Comment:
Can you add default value here as well? So it doesn't break for no reason.
```JS
const { groups, pagination, userIdentity, isLoading } = useSelector((
  { groupReducer: { groups, isLoading }} = { groupReducer: {} }
) => ({
```

Reply:
I would solve this on the groupReducer level. There should always be an initial state.

Reply:
Currently I have
```
// Initial State
export const groupsInitialState = {
  groups: {
    data: [],
    meta: {
      count: 0,
      limit: 10,
      offset: 0
    }
  },
  selectedGroup: { addRoles: {}},
  isLoading: false,
  isRecordLoading: false
};
```
in group-reducer.js, so it should not fail, should it? There are default values for everything.

Reply:
Then there should be no problem